homeassistant/components/shelly/__init__.py

```diff
@@ -220,6 +220,10 @@ def _async_device_updates_handler(self):
 
     async def _async_update_data(self):
         """Fetch data."""
+        if self.entry.data.get("sleep_period"):
+            # Sleeping device, no point polling it, just mark it unavailable
+            raise update_coordinator.UpdateFailed("Sleeping device did not update")
```

Comment:
We shouldn't raise, this will print a warning. Can we avoid subscribing to the data update coordinator completely?

Reply:
My intention was to print a warning here.

I think that we should want the same behavior for sleeping devices and non-sleeping devices.

Our current implementation for non sleeping devices is as follows:
We set the update interval at startup by what is declared by the device. If device does not update within `1.2 * interval` we try to poll it and if this fails we print a warning and mark it as unavailable.

If we want the same behavior for non-sleeping devices if we reach here we have a problem it means the device did not update within `1.2 * sleep interval` so I raise an exception similar to non-sleeping devices.

If we avoid subscribing to the data update coordinator and a sleeping device is dead, we have nothing to trigger a check to validate it is has not updated and it will stay available forever.

We can change our `available` property to report state based on time from last update and remove the polling completely here or remove only for non-sleeping devices and just return here without an exception.

Reply:
Ah right. This does indeed give the same behavior.